modules/ducktests/tests/checks/utils/check_marks.py

```diff
@@ -0,0 +1,240 @@
+        ]
+    )
+    def check_injection_pairs_fail(self, versions, version_prefix, global_args):
+        """
+        Check incorrect injecting with pairs of versions.
+        """
+        @ignite_versions(*versions, version_prefix=version_prefix)
+        def function(pair_1, pair_2):
+            return IgniteVersion(pair_1), IgniteVersion(pair_2)
+
+        with pytest.raises(Exception):
+            context_list = expand_function(func=function, sess_ctx=mock_session_ctx(global_args=global_args))
+
+            self._check_injection(context_list, versions=versions, global_args=global_args, pairs=True)
+
+    def check_with_others_marks(self):  # pylint: disable=R0201
+        """
+        Checks that ignite version parametrization works with others correctly.
+        """
+        @ignite_versions(str(DEV_BRANCH), str(V_2_8_1), version_prefix='ver')
+        @parametrize(x=10, y=20)
+        @parametrize(x=30, y=40)
+        def function_parametrize(ver, x, y):
+            return ver, x, y
+
+        @ignite_versions((str(DEV_BRANCH), str(V_2_8_1)), (str(V_2_8_1), str(V_2_7_6)), version_prefix='pair')
+        @matrix(i=[10, 20], j=[30, 40])
+        def function_matrix(pair_1, pair_2, i, j):
+            return pair_1, pair_2, i, j
+
+        @ignore(ver=str(DEV_BRANCH))
+        @ignite_versions(str(DEV_BRANCH), str(V_2_8_1), version_prefix='ver')
+        def function_ignore(ver):
+            return ver
+
+        context_list = expand_function(func=function_parametrize, sess_ctx=mock_session_ctx())
+        context_list += expand_function(func=function_matrix, sess_ctx=mock_session_ctx())
+        context_list += expand_function(func=function_ignore, sess_ctx=mock_session_ctx())
+
+        assert len(context_list) == 14
+
+        parametrized_context = list(filter(lambda x: x.function_name == function_parametrize.__name__, context_list))
+        assert len(parametrized_context) == 4
+        for ctx in parametrized_context:
+            args = ctx.injected_args
+            assert len(args) == 3
+            assert ctx.function() == (args['ver'], args['x'], args['y'])
+
+        matrix_context = list(filter(lambda x: x.function_name == function_matrix.__name__, context_list))
+        assert len(matrix_context) == 8
+        for ctx in matrix_context:
+            args = ctx.injected_args
+            assert len(args) == 4
+            assert ctx.function() == (args['pair_1'], args['pair_2'], args['i'], args['j'])
+
+        assert len(list(filter(lambda x: x.function_name == function_ignore.__name__, context_list))) == 2
+        assert len(list(filter(lambda x: x.ignore, context_list))) == 1
+
+    @staticmethod
+    def _check_injection(context_list, *, versions, global_args=None, pairs=False):
+        if global_args:
+            global_versions = global_args['ignite_versions']
+
+            if isinstance(global_versions, str):
+                assert not pairs
```

Comment:
As I understand it checks that test parameters are correct, but do not test decorator itself. So exception in this line can hide error in decorator implementation. For example, test expect exception but test parameters are wrong -> it leads to exception but do not test decorator. Am I correct?

Reply:
I don't understand you, I tried but no clue what are you talking about

Reply:
removed